### ISSUE TITLE
Function breaks when `group_by()` is done

### ISSUE THREAD

**Issue by AdrianLJones** [STAT545-UBC/Discussion#157]
I have a function very similar to the regression we did in class except that it does a polynomial regression and I can choose which columns of the dataset to regress.

``` r
poly_coeffs <- function (data, x, y, order, offset, raw = TRUE) {
   #This funciton fits input vector x at order = order to input vecotor y using R's standard lm() function.
    #It then returns the resulting coefficients using the broom::tidy

   fit <- lm(data[,y] ~ poly(data[,x] - offset, order, raw = raw))
   poly_coeffs <- tidy(fit)
}
```

It works fine on a single country

``` r
x <- "year"
y <- "lifeExp"
order <- 2
offset <- 1952
SA_test <- gapminder %>% filter(country == "South Africa") %>% do(poly_coeffs(., x, y, order, offset))
```

But gets upset when I try to `do` it to several countries

``` r
gap_sub <- gapminder %>% filter(country == "South Africa" | country == "France" | country == "Rwanda"| country == "Canada") %>% group_by(country) %>% do(poly_coeffs(., x, y, order, offset))
## Error in FUN(X, Y, ...) : non-numeric argument to binary operator 
```

Interestingly the single country also fails when I make gapminder a `tbl_df`  I think there might be something slightly different about the way dataframe column names are handled that prevents me using the `data.frame[ , column.name]` system of grabbing the column.  Is that right?  Any suggestions on other ways to define which column to use of a data frame inside a function that might work around this problem?

Thanks

Adrian

ps [here](https://github.com/STAT545-UBC/adrian_jones/blob/master/Homework/Homework04/Homework04_function.Rmd) is my current .rmd file for HW04 in case my description is insufficient,

**Comment by jennybc** [comment]
Can you rephrase your big `filter(...)` statement as `filter(country %in% c("South Africa", "France", "Rwanda", "Canada")`? That will be a step in the right direction.

**Comment by jennybc** [comment]
@AdrianLJones I took the liberty of reformatting your code to make it easier to read.

**Comment by AdrianLJones** [comment]
@jennybc Thanks for the help, I've ended up hard coding the x and y values as you suggested in class.
